File: src/MyServiceBus.AzureServiceBus/AzureServiceBusTransportException.cs
namespace MyServiceBus;

public sealed class AzureServiceBusTransportException : Exception
{
    public AzureServiceBusTransportException(string operation, string entityName, Exception innerException)
        : base($"Azure Service Bus operation '{operation}' failed for entity '{entityName}'.", innerException)
    {
        Operation = operation;
        EntityName = entityName;
    }

    public string Operation { get; }

    public string EntityName { get; }
}

File: src/Java/myservicebus-azure-service-bus/src/main/java/com/myservicebus/azure/servicebus/AzureServiceBusTransportException.java
package com.myservicebus.azure.servicebus;

public final class AzureServiceBusTransportException extends RuntimeException {
    private final String operation;
    private final String entityName;

    public AzureServiceBusTransportException(String operation, String entityName, Throwable cause) {
        super("Azure Service Bus operation '" + operation + "' failed for entity '" + entityName + "'.", cause);
        this.operation = operation;
        this.entityName = entityName;
    }

    public String getOperation() {
        return operation;
    }

    public String getEntityName() {
        return entityName;
    }
}
